firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/FirebaseMlException.java

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.ml.modeldownloader;
+
+import androidx.annotation.IntDef;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.android.gms.common.internal.Preconditions;
+import com.google.firebase.FirebaseException;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Represents an Exception resulting from an operation on a {@link FirebaseModelDownloader}. Error
+ * mappings should remain consistent with the original firebase_ml_sdk whenever possible.
+ */
+public class FirebaseMlException extends FirebaseException {
+  /** The operation completed successfully. FirebaseMLException will never have a status of OK. */
```

Comment:
If it will never have this status, why include it as a possibility in the type?
i.e. it seems that by doing it you're making this a runtime error as opposed to a compile-time error.

Reply:
I removed for now - I think it was meant as a default in the old code but I can just not allow that here.